src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,314 @@
+
+  // Properties
+  private drafTranslations: Array<string> = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasNameTranslationChange (): boolean {
+    return this.drafTranslations.length !== this.originalNameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return this.originalNameTranslations[index] === value
+      })
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.drafTranslations = [ ...this.originalNameTranslations ]
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private cancelNameTranslationCorrection () {
+    const nameTranslations = this.nameTranslations || []
+    const hasUnsavedData = this.drafTranslations.length !== nameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return nameTranslations[index] === value
+      })
+    if (hasUnsavedData) {
+      this.confirmUnsavedChanges()
+    } else {
+      this.isEditing = false
+    }
+  }
+
+  private confirmUnsavedChanges () {
+    // open confirmation dialog and wait for response
+    this.$refs.confirmTranslationDialog.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Name Translations. Do you want to save these changes?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Save changes',
+        no: null,
+        cancel: 'Cancel without saving'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.setNameTranslations()
+      }
+    }).catch(() => {
+      this.onNameTranslationsPropValueChanged()
+      this.isEditing = false
+    })
+  }
+
+  /** Add or update a name translation
+   *
+   * @param name The name to add
+   */
+  private addName (name: string): void {
+    // Handle name translation adds or updates
+    this.editIndex > -1
+      ? this.drafTranslations[this.editIndex] = name
+      : this.drafTranslations.push(name)
+
+    // Emit haveChanges
+    // this.emitHaveChanges(this.hasNameTranslationChange)
+
+    this.cancelNameTranslation()
+  }
+
+  /** Pass an index of the name translation to be edited
+   *
+   * @param index Index number of the name translation to edit
+   */
+  private editNameTranslation (index: number): void {
+    this.editingNameTranslation = this.drafTranslations[index]
+    this.editIndex = index
+    this.isAddingNameTranslation = true
+  }
+
+  /** Remove a name translation
+   *
+   * @param index Index number of the name translation to remove
+   */
+  private removeNameTranslation (index: number): void {
+    this.drafTranslations.splice(index, 1)
+
```

Comment:
Same here. Will you be able to populate removed translation without another level of comparison in the template mixin?

Reply:
Ah yes, normally you would keep track of which translations were deleted and then filter them out in the Filing Template Mixin.
But here, you just delete the translation (no Undo), right?

Reply:
^^^ Actually, Name Translations are not done this way. See the schema. There is a filing section for new, modified and ceased.

(I don't know how unchanged translations are stored...?)

Reply:
I think this conversation is superceded by Lekshmi's (and my) comments below.